### ISSUE TITLE
Declaration of war ends in war with alliance partner

### ISSUE THREAD

**Issue by JHCD** [chaorace/cqui#371]
Initial position:
I had an alliance with two AI player. 

After a third AI player (Japan) declared war, the game was unstable. So it looked like I was in war with one of the AI players (China) too. But there was no declaration of war from China.

![error_war_china](https://cloud.githubusercontent.com/assets/12496340/23604679/fb145682-0259-11e7-8a76-5385f3429c11.jpg)

China was no agressor, but was defending when I moved units through his country.
That was not the first time I saw this issue and only when I played games mit CQUI enabled.

**Comment by JHCD** [comment]
Could there be a dependency to issue #372?

**Comment by chaorace** [comment]
I think this issue is actually happening because the game's "hostile" tag is quite generous and includes certain units always, like spies. This coloration is still a bug though since it's quite misleading. I'll add a check for this case

**Comment by JHCD** [comment]
Thats not enough...
1. I could kill all units from that country
2. Hostile units and cities try to kill my units (defence)

There is only no active war...

**Comment by JHCD** [comment]
You got savegame and logs by eMail

**Comment by chaorace** [comment]
Yep, message received, I'll be testing it soon as a batch with a bunch of other saves I've been sent for other issues

**Comment by ricanuck** [comment]
Cannot replicate. Never saw it happening.

**Comment by SpaceOgre** [comment]
Saw the opposite happening yesterday on Quill18's stream. Sumeria declared war on him but not all enemy units where red tinted.

**Comment by JHCD** [comment]
Chris has the relevant savegame... so he could check it too.

**Comment by ricanuck** [comment]
Because he is using an old version of the mod. I fixed this last week, working perfectly now. Somebody should tell him to update, he will be a good promoter of the mod, and now that it has some more nice perks, he will love it.

**Comment by chaorace** [comment]
@quill18 Heyo, you may want to considering updating to the latest CQUI release to fix that hostile unit highlighting issue you've been experiencing. There's also a workshop version available now for keeping up to date automatically!

As for the issue in the topic, I'll test the save soon to see if I can replicate

**Comment by AtomicDeke** [comment]
I'm running into this same issue now. I was allied with Gorgo, Saladin declared war on me, and then Gorgo was registering as hostile. I have since peaced out with Saladin, but Gorgo will still attack me if I wander near her territory, and every time I become a city-state's new suzerain, both Saladin and Gorgo will declare war on it. 

There's another wrinkle that seems to be related. When I first discovered that Gorgo was treating my units as hostile, she bombarded my battleship. I haven't been able to repair it in my own territory since then. I'm guessing the game's not recognizing it as hurt since the UI's not showing any hostilities between me and Gorgo?

**Comment by chaorace** [comment]
@AtomicDeke I suspect that this may be a rare vanilla issue, could you provide a save file so I can test this?

**Comment by AtomicDeke** [comment]
Sure thing, I emailed you the save file.
